capsules/src/net/lowpan_fragment.rs

```diff
@@ -0,0 +1,863 @@
+    src_pan: Cell<PanID>,
+    dst_pan: Cell<PanID>,
+    src_mac_addr: Cell<MacAddress>,
+    dst_mac_addr: Cell<MacAddress>,
+    security: Cell<Option<(SecurityLevel, KeyId)>>,
+    dgram_tag: Cell<u16>,
+    dgram_size: Cell<u16>,
+    dgram_offset: Cell<usize>,
+    fragment: Cell<bool>,
+    compress: Cell<bool>,
+    client: Cell<Option<&'static TransmitClient>>,
+
+    next: ListLink<'a, TxState<'a>>,
+}
+
+impl<'a> ListNode<'a, TxState<'a>> for TxState<'a> {
+    fn next(&'a self) -> &'a ListLink<TxState<'a>> {
+        &self.next
+    }
+}
+
+impl<'a> TxState<'a> {
+    /// TxState::new
+    /// ------------
+    /// This constructs a new, default TxState struct.
+    pub fn new() -> TxState<'a> {
+        TxState {
+            packet: TakeCell::empty(),
+            src_pan: Cell::new(0),
+            dst_pan: Cell::new(0),
+            src_mac_addr: Cell::new(MacAddress::Short(0)),
+            dst_mac_addr: Cell::new(MacAddress::Short(0)),
+            security: Cell::new(None),
+            dgram_tag: Cell::new(0),
+            dgram_size: Cell::new(0),
+            dgram_offset: Cell::new(0),
+            fragment: Cell::new(false),
+            compress: Cell::new(false),
+            client: Cell::new(None),
+            next: ListLink::empty(),
+        }
+    }
+
+    /// TxState::set_transmit_client
+    /// ----------------------------
+    /// Sets the TransmitClient callback field of the respective TxState struct.
+    pub fn set_transmit_client(&self, client: &'static TransmitClient) {
+        self.client.set(Some(client));
+    }
+
+    fn is_transmit_done(&self) -> bool {
+        self.dgram_size.get() as usize <= self.dgram_offset.get()
+    }
+
+    fn init_transmit(&self,
+                     src_mac_addr: MacAddress,
+                     dst_mac_addr: MacAddress,
+                     packet: &'static mut [u8],
+                     packet_len: usize,
+                     security: Option<(SecurityLevel, KeyId)>,
+                     fragment: bool,
+                     compress: bool) {
+
+        self.src_mac_addr.set(src_mac_addr);
+        self.dst_mac_addr.set(dst_mac_addr);
+        self.fragment.set(fragment);
+        self.compress.set(compress);
+        self.security.set(security);
+        self.packet.replace(packet);
+        self.dgram_size.set(packet_len as u16);
+    }
+
+    // Takes ownership of frag_buf and gives it to the radio
+    fn start_transmit(&self,
+                      dgram_tag: u16,
+                      mut frag_buf: &'static mut [u8],
+                      radio: &Mac,
+                      ctx_store: &ContextStore)
+                      -> Result<ReturnCode, (ReturnCode, &'static mut [u8])> {
+        self.dgram_tag.set(dgram_tag);
+        let ip6_packet_option = self.packet.take();
+        if ip6_packet_option.is_none() {
+            return Err((ReturnCode::ENOMEM, frag_buf));
+        }
+        let ip6_packet = ip6_packet_option.unwrap();
```

Comment:
Shouldn't use `unwrap` if avoidable. In this case, i would restructure it like this:

```rust
match ip6_packet_option {
    None => return Err((ReturnCode::ENOMEM, frag_buf)),
    Some(ip6_packet) => {
         ...rest of method...
    }
}
```